### ISSUE TITLE
Update enum property doesn't work

### ISSUE THREAD

**Issue by xuzhg** [OData/WebApi#742]
for example:

1. Request method: Patch
2. Request Uri:  `~/Customers(1)/FavoriteColor`
3. Request Payload:
```json
{"@odata.type":"#NS.Color","value":"Blue"}
```
4. Action in controller:

```
[HttpPatch]
public IHttpActionResult PatchToFavoriteColor(int key, [FromBody] Color color)
{
    Assert.Equal(Color.Blue, color);
    return Ok();
}
```

The above assert should be successful. But, it's not.

### Assemblies affected
Web API OData 5.9.1， 7.0  

### Expected result
The above assert should be successful. But, it's not.

### Actual result
It's failed.

### Additional details

https://github.com/OData/WebApi/blob/c44ce5547b1b44474bfa2d1208c68b90fa401556/OData/src/System.Web.OData/OData/Formatter/Deserialization/ODataEnumDeserializer.cs#L44

It should call: 
```
return ReadInline(property.Value, edmType, readContext);
```

**Comment by xuzhg** [comment]
fixed at:  9a00734

**Comment by xuzhg** [comment]
included in v6.0
